engine/table/src/main/java/io/deephaven/engine/util/PythonScopeJpyImpl.java

```diff
@@ -121,65 +203,66 @@ public Class<?> getReturnType() {
-        numpyType2JavaClass.put('l', long.class);
-        numpyType2JavaClass.put('h', short.class);
-        numpyType2JavaClass.put('f', float.class);
-        numpyType2JavaClass.put('d', double.class);
-        numpyType2JavaClass.put('b', byte.class);
-        numpyType2JavaClass.put('?', boolean.class);
-    }
+        public boolean isVectorizable() {
+            return vectorizable;
+        }
 
-    private static CallableWrapper parseNumbaVectorized(PyObject pyObject, String numbaFuncTypes) {
-        // the 'types' field of a numba vectorized function takes the form of
-        // '[parameter-type-char*]->[return-type-char]',
-        // eg. [ll->d] defines two int64 (long) arguments and a double return type.
+        public void setVectorizable(boolean vectorizable) {
+            this.vectorizable = vectorizable;
+        }
 
-        char numpyTypeCode = numbaFuncTypes.charAt(numbaFuncTypes.length() - 1);
-        Class<?> returnType = numpyType2JavaClass.get(numpyTypeCode);
-        if (returnType == null) {
-            throw new IllegalArgumentException(
-                    "numba vectorized functions must have an integral, floating point, or boolean return type.");
+        public void clearArgs() {
+            this.args.clear();
         }
 
-        List<Class<?>> paramTypes = new ArrayList<>();
-        for (char numpyTypeChar : numbaFuncTypes.toCharArray()) {
-            if (numpyTypeChar != '-') {
-                Class<?> paramType = numpyType2JavaClass.get(numpyTypeChar);
-                if (paramType == null) {
-                    throw new IllegalArgumentException(
-                            "parameters of numba vectorized functions must be of integral, floating point, or boolean type.");
-                }
-                paramTypes.add(numpyType2JavaClass.get(numpyTypeChar));
-            } else {
-                break;
-            }
+        public void addArg(Pair<?, ?> arg) {
+            this.args.add(arg);
+        }
+
+        public ArrayList<Pair<?, ?>> getArgs() {
+            return this.args;
+        }
+
+        public void setArgs(ArrayList<Pair<?, ?>> args) {
+            this.args = args;
+        }
+
+        public Class<?>[] getArgTypes() {
+            return argTypes;
         }
 
-        if (paramTypes.size() == 0) {
-            throw new IllegalArgumentException("numba vectorized functions must have at least one argument.");
+        public void setArgTypes(Class<?>[] argTypes) {
+            this.argTypes = argTypes;
         }
-        return new NumbaCallableWrapper(pyObject, returnType, paramTypes);
+    }
+
+    public static CallableWrapper vectorizeCallable(PyObject pyObject) {
+        PyObject vectorized = dh_table_module.call("dh_vectorize", pyObject);
+        CallableWrapper vectorizedCallableWrapper = new CallableWrapper(vectorized);
+        vectorizedCallableWrapper.parseSignature();
+        return vectorizedCallableWrapper;
+    }
```

Comment:
There is some repeated code elsewhere that calls this method and then sets args and argtypes.  Is it worth having a non-static method that returns a vectorized function with these values set?  Seems like it could simplify the other code.  If this is done, this method can probably become private.